README.md

```diff
@@ -38,105 +52,212 @@
+
+Slave images are created from Dockerfiles.   
+
+````
+├── config
+│   └── s2i
+│       ├── create-containers.sh
+│       ├── jenkins
+│       │   ├── continuous-infra-create-openshift-project
+│       │   ├── continuous-infra-jenkins-master-s2i-template.yaml
+│       │   ├── continuous-infra-slave-template.yaml
+│       │   ├── master
+│       │   │   ├── configuration
+│       │   │   │   ├── hudson.plugins.ircbot.IrcPublisher.xml
+│       │   │   │   ├── hudson.tasks.Maven.xml
+│       │   │   │   ├── init.groovy
+│       │   │   │   └── jobs
+│       │   │   │       ├── fedmsg-test
+│       │   │   │       │   └── config.xml
+│       │   │   │       └── maven-test
+│       │   │   │           └── config.xml
+│       │   │   └── plugins.txt
+│       │   └── slave
+│       │       └── Dockerfile
+
+````
+
+##### Pipeline Shared Libraries
+
+We have shared libraries that are used in this repo/project and we also use other shared libraries that we contribute to along with the CentOS Community<br>
+[cico-pipeline-library](https://github.com/CentOS/cico-pipeline-library)
+<br>
 
-Setup and invoking of tests will be done using ansible as outlined in [Invoking Tests Using Ansible](https://fedoraproject.org/wiki/Changes/InvokingTestsAnsible)
+The libraries in this repo/project follows the [Jenkins community shared library structure](https://jenkins.io/doc/book/pipeline/shared-libraries/):
 
-Setup and invoking of tests will be done using ansible as outlined in Invoking Tests Using Ansible
+````
+├── src
+│   └── org
+│       └── centos
+│           └── pipeline
+│               └── PipelineUtils.groovy
+└── vars
+    └── pipelineUtils.groovy
+````
 
-We assume that there will be tests available to execute otherwise there is nothing besides building, composing, and configuring an Openshift cluster to validate the OStrees and images.  There is currently tests available for many of the Atomic components as well as the Kubernetes e2e conformance tests.
+### CI the CI-Pipeline - Inception
 
-The CI-Pipeline can be defined using Jenkins Job Builder short-term and Jenkins Pipeline (Jenkinsfile) long-term.  This will be the Jenkins 2.0 Pipeline that is already integrated in Openshift [(Openshift Pipelines Deep Dive)](https://blog.openshift.com/openshift-3-3-pipelines-deep-dive/).
+Since we can easily integrate into github then we can use our own pipeline to validate our stage containers, pipeline code, and shared libraries.  
+ 1. We inspect the change.log in the pipeline using the [Jenkins declaritive pipeline DSL](https://github.com/CentOS-PaaS-SIG/ci-pipeline/blob/master/JenkinsfileStageTrigger).
+ 2. If stage containers are touched then they are built in Openshift and validate
+ 3. If the stage container passes validation it is tagged stable and promoted to production in the pipeline
+ 4. If a container and/or pipeline code and/or shared libraries are changed the change is run through th pipeline for validation
```

Comment:
are changed, then the change

also typo -> 'th'

Reply:
good catch